src/main/java/ratpack/zipkin/internal/DefaultServerTracingHandler.java

```diff
@@ -82,6 +84,20 @@ public String getPath() {
     public Headers getHeaders() {
       return request.getHeaders();
     }
+
+    @Override
+    public String getUrl() {
+      PublicAddress publicAddress = request.get(Context.class).get(PublicAddress.class);
+      return publicAddress.builder()
+                          .path(request.getPath())
+                          .params(request.getQueryParams())
+                          .build().toString();
```

Comment:
I'm assuming that the URL we want is whatever the client "sees". [`PublicAddress`](https://ratpack.io/manual/current/api/ratpack/server/PublicAddress.html) is the Ratpack class that is used to create absolute URLs and redirects, so figured this would be appropriate.

Reply:
yep